### ISSUE TITLE
Color Picker in Network Configuration is misplaced when parent scrolled in dialog

### ISSUE THREAD

**Issue by ssadams11** [almende/vis#1145]
I'm using the configuration panel inside a jquery dialog, which works great, but the color picker opens way off target , out of sight, since the dialog content gets scrolled.

Could we show color picker on top of dialog as opposed to inside it and ensure its location is in view?
![image](https://cloud.githubusercontent.com/assets/5204570/9013659/6170c8b6-378b-11e5-81e5-dafcc226f872.png)
![image](https://cloud.githubusercontent.com/assets/5204570/9013669/755a78ae-378b-11e5-8705-18c956f537c1.png)

**Comment by felixhayashi** [comment]
I am also using the picker in a dialog and I can confirm this issue.

**Comment by AlexDM0** [comment]
Hi,

I've put it in the body now. This has been pushed to develop.

Could you guys test it?

Regards

**Comment by ssadams11** [comment]
Tried it but no joy.  Picker appears in middle of dialog content (the config panel), even if scrolled to top of panel.  Here's my dialog code:
```coffeescript
$('.all-settings').click (evt)->
    $('#vis-config').dialog({title: 'Visualization Options',resizable: true,width:500,height:300})
```
And the setup:
```coffeescript
 container = document.getElementById 'mynetwork'
  config = document.getElementById 'vis-config'
  $(config).hide()
  options =
    interaction:
      hover: true
      navigationButtons: true
      multiselect: true
    manipulation: true
    configure:
      enabled: true
      container: config
      showButton: true
    physics: true
    nodes:
      hidden: false
      color:
        hover:
          border: 'yellow'
          background: 'darkblue'
        highlight:
          border: 'red'
          background: 'yellow'
    edges:
      hidden: false
      arrows:
        to:
          enabled: true
          scaleFactor: 0.5
      color:
        highlight:'red'
        hover:'darkblue'
```

**Comment by AlexDM0** [comment]
I'm... not sure what to make of it. I currently use https://developer.mozilla.org/en/docs/Web/API/Element/getBoundingClientRect to get the top and left positions. It should get the top and left relative to the viewport. If you scroll down in a div then yes, it could go wrong. Does it never align correctly?

Regards

**Comment by ssadams11** [comment]
https://jsfiddle.net/ssadams/er84oj6y/

Interesting, the fiddle above is using the latest CDN of Vis.js, not your dev branch, but it has different behavior than my app.  In mine the ColorPicker is inside the dialog window.  Its on top like we want here, but the placement is still off.  I've also hacked the css given the conflicts between Vis.js and Bootstrap.

![screen shot 2015-08-07 at 6 03 08 pm](https://cloud.githubusercontent.com/assets/5204570/9146879/9642e788-3d2e-11e5-848f-144ee6228516.png)

Here is the screens from my app in similar situation:
First is after I click the color for Node border color.  The second image is where the picker displays, near edge stroke width.
![screen shot 2015-08-07 at 6 06 08 pm](https://cloud.githubusercontent.com/assets/5204570/9146972/52348b86-3d2f-11e5-9e2e-b97275360330.png)
![screen shot 2015-08-07 at 6 06 57 pm](https://cloud.githubusercontent.com/assets/5204570/9146974/57be03d4-3d2f-11e5-84c0-87fe6233e8c1.png)

**Comment by AlexDM0** [comment]
Does anyone have a suggestion how we can setup CSS to ensure that the color picker is always on top and will correctly match the configuration popup?

**Comment by felixhayashi** [comment]
Hi Alex,

to be honest, I have no idea :( Maybe it is better to remove the speech bubble triangle and just say to the visplugin user: place the configurator statically somewhere using css (left top).

I'll experiment a bit the next days/week with the configurator but if you do not hear anything from me, that probably means I did not find a solution for this.

-Felix

**Comment by AlexDM0** [comment]
Hi Felix,

I take it you havent found a solution? :/

Regards

**Comment by felixhayashi** [comment]
Hmm an idea would be to make the color picker spawn inside the clicked `vis-config-colorBlock` div instead of simply placing it inside the `vis-configuration-wrapper` container and switching `display: block/none`. Then `vis-config-colorBlock` needs to be set to `position: relative;` and once clicked, the picker's `insertTo()` is called with the color block div passed.

What do you think?

**Comment by AlexDM0** [comment]
Hmm that could work... Feel like trying to implement it? ;)

**Comment by felixhayashi** [comment]
Ok, I'll give it a try.

**Comment by AlexDM0** [comment]
Hi,

Should be fixed by Felix's pull.

Will be released friday!

Cheers

**Comment by felixhayashi** [comment]
> Will be released friday!

:+1:

**Comment by felixhayashi** [comment]
@ssadams11 maybe you can checkout the develop branch to see if this is also working for you (should be the case).

**Comment by ssadams11** [comment]
Will do!


Sam S. Adams, Cognitive Algorithms
IBM Distinguished Engineer, IBM Research
Mobile: 919-696-6064, email: ssadams@us.ibm.com
Assistant: Tonja Ross, 404-348-9086, Tie-Line: 824-9086,
tonja_ross@us.ibm.com
<<Hebrews 11:6, Proverbs 3:5-6, Romans 1:16-17, 1 Corinthians 1:10>>

**Comment by AlexDM0** [comment]
Ths has been released in 4.12
